src/openrct2/scripting/ScriptEngine.cpp

```diff
@@ -1127,6 +1134,47 @@ void ScriptEngine::SaveSharedStorage()
     }
 }
 
+#    ifndef DISABLE_NETWORK
+void ScriptEngine::AddSocket(const std::shared_ptr<ScSocketBase>& socket)
+{
+    _sockets.push_back(socket);
+}
+#    endif
+
+void ScriptEngine::UpdateSockets()
+{
+#    ifndef DISABLE_NETWORK
+    // Use simple for i loop as Update calls can modify the list
+    for (size_t i = 0; i < _sockets.size(); i++)
+    {
+        _sockets[i]->Update();
+        if (_sockets[i]->IsDisposed())
```

Comment:
i can be potentially set to -1 if _sockets[0]->IsDisposed()==true

Reply:
I don't think it can.

Reply:
It will underflow temporarily but I don't think thats an issue here, its a bit odd tho.

Reply:
This is a pretty common way of removing items while iterating through. Even if you use an iterator (which I can't because update modifies the collection), you still typically subtract 1 to counter act the loop modifier.